x-pack/legacy/plugins/maps/public/plugin.ts

```diff
@@ -20,23 +23,31 @@ import { setLicenseId, setInspector } from './kibana_services';
 export type MapsPluginSetup = ReturnType<MapsPlugin['setup']>;
 export type MapsPluginStart = ReturnType<MapsPlugin['start']>;
 
+interface MapsPluginSetupDependencies {
+  __LEGACY: any;
+  np: {
+    licensing?: LicensingPluginSetup;
+    home: HomePublicPluginSetup;
+  };
+}
+
 /** @internal */
 export class MapsPlugin implements Plugin<MapsPluginSetup, MapsPluginStart> {
-  public setup(core: any, plugins: any) {
-    const {
-      __LEGACY: { uiModules },
-      np: { licensing },
-    } = plugins;
-
+  public setup(
+    core: any,
+    { __LEGACY: { uiModules }, np: { licensing, home } }: MapsPluginSetupDependencies
+  ) {
     uiModules
       .get('app/maps', ['ngRoute', 'react'])
       .directive('mapListing', function(reactDirective: any) {
         return reactDirective(wrapInI18nContext(MapListing));
       });
 
     if (licensing) {
-      licensing.license$.subscribe(({ uid }: { uid: string }) => setLicenseId(uid));
+      licensing.license$.subscribe(({ uid }) => setLicenseId(uid));
```

Comment:
is removal of the type-def for cosmetic purposes only?

Reply:
Hey, yeah.
Since the `plugins` has a strict type `MapsPluginSetupDependencies`, the `licensing` dep is already typed.